subprojects/file-watching/src/main/java/org/gradle/internal/watch/registry/impl/HierarchicalFileWatcherUpdater.java

```diff
@@ -61,7 +64,8 @@
 public class HierarchicalFileWatcherUpdater implements FileWatcherUpdater {
     private static final Logger LOGGER = LoggerFactory.getLogger(HierarchicalFileWatcherUpdater.class);
 
-    private final Set<Path> watchedHierarchies = new HashSet<>();
+    private final Set<Path> watchedRoots = new HashSet<>();
+    private FileHierarchySet watchedHierarchies = DefaultFileHierarchySet.of();
```

Comment:
I think we should extract these two to a new type, similar to `WatchableHierarchies`.

Reply:
I moved both fields to `WatchedHierarchies`.